plugins/commands/ps/command.rb

```diff
@@ -0,0 +1,116 @@
+require "optparse"
+
+require_relative "../../communicators/winrm/helper"
+
+module VagrantPlugins
+  module CommandPS
+    class Command < Vagrant.plugin("2", :command)
+      def self.synopsis
+        "connects to machine via powershell remoting"
+      end
+
+      def execute
+        options = {}
+
+        opts = OptionParser.new do |o|
+          o.banner = "Usage: vagrant ps [-- extra ps args]"
+
+          o.separator ""
+          o.separator "Options:"
+          o.separator ""
+
+          o.on("-c", "--command COMMAND", "Execute a powershell command directly") do |c|
```

Comment:
Yeah I'm again sticking to the same language used in the `vagrant ssh` command and the functionality is exactly the same. If a `--command` is supplied, it is executed on the vagrant box, otherwise one is dropped into an interactive powershell session.